torch/csrc/api/src/nn/init.cpp

```diff
@@ -53,12 +105,16 @@ double calculate_kaiming_std(
 }
 } // namespace
 
-double calculate_gain(Nonlinearity nonlinearity, double param) {
-  if (nonlinearity == torch::nn::init::Nonlinearity::Tanh) {
+double calculate_gain(NonlinearityType nonlinearity, double param) {
+  // Support for `torch::nn::init::Nonlinearity` is deprecated and will be removed in 1.5.
+  if (c10::get_if<Nonlinearity>(&nonlinearity)) {
+    nonlinearity = _compute_nonlinearity_type(c10::get<Nonlinearity>(nonlinearity));
+  }
+  if (c10::get_if<enumtype::kTanh>(&nonlinearity)) {
     return 5.0 / 3.0;  // NOLINT
-  } else if (nonlinearity == torch::nn::init::Nonlinearity::ReLU) {
+  } else if (c10::get_if<enumtype::kReLU>(&nonlinearity)) {
     return std::sqrt(2.0);  // NOLINT
-  } else if (nonlinearity == torch::nn::init::Nonlinearity::LeakyReLU) {
+  } else if (c10::get_if<enumtype::kLeakyReLU>(&nonlinearity)) {
     return std::sqrt(2.0 / (1 + pow(param, 2)));  // NOLINT
   }
```

Comment:
Hmm. If we ever add a new variant to the enum, will we then get an error saying we didn't handle all cases yet? If not, we should add a fallthrough case here...

Reply:
Yeah we should have a fallthrough case here - I will do it as part of the work for `torch::nn::init::calculate_gain` parity.